### ISSUE TITLE
"Still compiling" indefinitely

### ISSUE THREAD

**Issue by webdif** [brunch/brunch#1594]
### Description

![brunch](https://cloud.githubusercontent.com/assets/1285888/21388108/18469d10-c77b-11e6-85aa-d9036eab6f54.png)

### Environment

1. Brunch: 2.9.1
2. Node: 6.2.2
3. NPM: 3.9.5
4. Operating system: macOS 10.12.2

---

I did not noticed this behaviour until this morning, everything was fine before… Maybe someone has already encountered this problem?

**Comment by shvaikalesh** [comment]
Hey @webdif, could you please share your `brunch-config` and plugins you are using? I am interested in `onCompile` hooks they define. Thanks in advance.

**Comment by webdif** [comment]
Hello @shvaikalesh!
Here is my `brunch-config.coffee`:

```coffee
exports.config =
	###
	# Plugins options
	###
	plugins:
		# Css
		stylus:
			paths: [
				'app/css'
				'src/components'
			]
		postcss:
			processors: [
				require('autoprefixer')
			]

		# Js
		babel:
			presets: ['es2015', 'react']
			plugins: [
				'transform-decorators-legacy'
				'transform-object-rest-spread'
				'transform-class-properties'
			]
		uglify:
			compress:
				drop_console: true
				drop_debugger: true

		# Img
		imageoptimizer:
			smushit: false
			path: 'images'

		# Server
		autoReload:
			enabled: false
		browserSync:
			enabled: true
			browser: 'Google Chrome Canary'
			logSnippet: false
			ghostMode: false

	###
	# Builds
	###
	files:
		javascripts:
			joinTo: 'js/app.js'

		stylesheets:
			joinTo: 'css/app.css'
			order:
				before: /^app\/css\/utils\/reset\.styl$/

	paths:
		public: '../web/assets'

	# modules:
	# 	autoRequire:
	# 		'js/app.js': ['src/client']

	conventions:
		assets: /assets\/(files|favicons\/(default|colors))/
		ignored: [
			/\/_/
			/\/css\/helpers/
		]
```

**Comment by shvaikalesh** [comment]
Could you please retry with Brunch `2.8.2`?

**Comment by webdif** [comment]
Same problem with `2.8.2`.

In fact, I had v2.8, and try to update brunch to 2.9 before posting here, in case of a fix have been made, but no luck 😕

**Comment by shvaikalesh** [comment]
OK, could you please try disabling `imageoptimizer` plugin?

**Comment by webdif** [comment]
Oooh, I just tried to reinstall `2.8.2`, and `brunch build`instead of `brunch watch`, and indeed it works!
I'm working on several projects, I may have been wrong and was already in `2.9` on this project.

But, I'm still having an issue with `brunch watch`, I don't know if it's related:
```
/Applications/MAMP/frontend/node_modules/brunch/lib/fs_utils/file_list.js:224
        if (error) throw new Error(formatError('Reading', error));
                   ^

Error: undefined of Error: EISDIR: illegal operation on a directory, read failed. 
  at error (/Applications/MAMP/htdocs/frontend/node_modules/brunch/lib/fs_utils/file_list.js:224:26)
  at /Applications/MAMP/htdocs/frontend/node_modules/fcache/index.js:19:23
  at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:424:12)
```

**Comment by webdif** [comment]
Uninstalling `imageoptmizer-brunch` seems to have no effect.

**Comment by webdif** [comment]
Don't know why this has not show up before, but now, I see in console `warn: StylusCompiler is taking long to process`. But I still have no clue what is going on, where and why… 😢

**Comment by shvaikalesh** [comment]
Hmm, are you sure u have `stylus-compiler` version defined as `^2`?

**Comment by webdif** [comment]
What is `stylus-compiler`? I normally just use `stylus`.
Btw, I've just updated `stylus` globally, with no effect…

**Comment by webdif** [comment]
I see that the `stylus-brunch` module is using a `StylusCompiler` class, is this you're referring to? In that case, my `stylus-brunch` is up to date, v2.8.0

**Comment by webdif** [comment]
I found a solution: it was a stupid error in an `import`… 😒 
But this is definitely an issue, as no error is shown, even with the debug option `-d`.

**Comment by shvaikalesh** [comment]
Hey @webdif, right, `stylus-brunch` is what I had on my mind. Sorry.

I believe https://github.com/es128/progeny/pull/32 should fix that.

**Comment by webdif** [comment]
Nice 👍 
I closed this issue then.

**Comment by shvaikalesh** [comment]
Progeny with fix was published as `0.11.1`

**Comment by webdif** [comment]
Nice! Brunch package will be updated on npm soon?
